### DISCUSSION TITLE
Future of the Keptn Web UI

### DISCUSSION THREAD

**Discussion by oleg-nenashev** [keptn/keptn#9051] [General]
As we discussed at the previous community meetings, there are concerns about long-term sustainability of the Keptn Bridge - a Web UI for Keptn.

In the previous conversations in the community, I believe we have a consensus that we would like to move to a headless solution with proper APIs and endpoints that would allow connecting an external interface, for example built on the top of Grafana or Backstage. 

Any feedback from Keptn users and developers would be welcome!

## Problems

- Old technology stack based on Angular. It is hard to develop and extend
- Huge backlog that needs a lot of involvement from the contributors to clean up: https://github.com/keptn/keptn/milestone/31
- Known performance issues. Some of them were addressed in Keptn 015...0.19, but there are still some known issues in the backlog
- UX issues for some controls, especially for reporting

## Possible Solutions

- Headless Keptn: Provide open API in Keptn so that external visualization tools could be connected, in particular Grafana or Backstage. Remove the UI from the Keptn core deliverables
  -  NOTE: It can be done with or without detaching the current interface as a separate project, or creating a new implementation
  - One could also delegate the Web UI to the observability provider if all data is streamed back there from Keptn 
- Update the Keptn bridge to the new technology stack, most likely by reimplementing it as reusable React or Vue components so that they could be used in external projects

**Comment by AloisReitbauer** [comment]
I am all for moving towards a headless approach for a number of reasons:
* Most users interact via the API already today. The total number of Bridge users is low.
* We are overall moving towards a composable tooling approach following the Linux paradigm. This makes building a UI pretty much impossible

I think this  also offer the opportunity to develop solutions on top of Keptn components. 

Having said all this, we need to define a clear path for existing users, how they can keep getting the information they are getting today in headless mode.
